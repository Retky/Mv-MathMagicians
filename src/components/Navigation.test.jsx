import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NavBar from './Navigation';

describe('Navbar component', () => {
  it('renders "Math Magicians"', () => {
    // Arrange
    render(
      <BrowserRouter><NavBar /></BrowserRouter>,
    );

    // Assert
    const headingElement = screen.getByText('Math Magicians');
    expect(headingElement).toBeInTheDocument();
  });

  it('renders component to match the previous snapshot', () => {
    // Assert
    const tree = renderer.create(<BrowserRouter><NavBar /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
