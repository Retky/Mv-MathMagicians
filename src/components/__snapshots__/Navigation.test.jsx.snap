// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders component to match the previous snapshot 1`] = `
<header>
  <h1>
    Math Magicians
  </h1>
  <ul
    className="navMenu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/calculator"
      onClick={[Function]}
    >
      Calculator
    </a>
    <a
      href="/quote"
      onClick={[Function]}
    >
      Quote
    </a>
  </ul>
</header>
`;
